superset/viz.py

```diff
@@ -439,14 +439,26 @@ def cache_key(self, query_obj: QueryObjectDict, **extra: Any) -> str:
 
     def get_payload(self, query_obj: Optional[QueryObjectDict] = None) -> VizPayload:
         """Returns a payload of metadata and data"""
+
         self.run_extra_queries()
         payload = self.get_df_payload(query_obj)
 
         df = payload.get("df")
+
+        # Check incompatible filters. Probably a better spot for this.
+        filters = self.form_data.get("filters")
```

Comment:
Make sure we add in Adhoc filters and Extra filters

Reply:
Potential refactor work for backend conversion of simple --> complex --> simple filters (ask @villebro)